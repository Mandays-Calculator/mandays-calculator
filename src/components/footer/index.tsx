export { default as Footer } from "./Footer";
export { StyledFooterContainer, StyledFooterText } from "./styles";
