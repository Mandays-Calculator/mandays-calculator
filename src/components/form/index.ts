export * from "./checkbox";
export * from "./checkbox-group";
export * from "./text-field";
export * from "./date-picker";
