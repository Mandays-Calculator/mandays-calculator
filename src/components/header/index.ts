export {
  StyledAppBar,
  StyledToolBarContainer,
  StyledSearchSelect,
  popperStyle,
  StyledProjectTitle,
} from "./styles";
