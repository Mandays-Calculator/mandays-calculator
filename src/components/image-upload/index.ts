export { default as ImageUpload } from "./ImageUpload";
export * from "./styles";
