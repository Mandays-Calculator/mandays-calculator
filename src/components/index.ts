export * from "./form/text-field";
export * from "./form/Form";
export * from "./title";
