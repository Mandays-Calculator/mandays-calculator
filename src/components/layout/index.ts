export { default as Layout } from "./Layout";
export {
  StyledLayoutContainer,
  StyledLayoutContent,
  ContentContainer,
} from "./styles";
