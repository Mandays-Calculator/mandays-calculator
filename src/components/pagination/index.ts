export { default as Pagination } from "./Pagination";
export * from "./styles";
