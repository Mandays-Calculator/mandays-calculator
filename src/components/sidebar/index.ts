export {
  DrawerHeader,
  StyledItemText,
  StyledListItem,
  StyledListItemIcon,
  StyledDrawer,
  StyledList,
} from "./styles";

export { default as Sidebar } from "./Sidebar";
