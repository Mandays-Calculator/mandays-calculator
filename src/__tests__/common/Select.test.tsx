import { render } from "@testing-library/react";
import { Select } from "~/components";

const options: SelectObject[] = [
  {
    label: "Test 1",
    value: "test1",
  },
  {
    label: "Test 2",
    value: "test2",
  },
];

describe("Select Component", () => {
  it("renders select without crashing", () => {
    const { getByTestId, asFragment } = render(
      <Select
        name="test-select"
        options={options}
      />
    );
    const selectElement = getByTestId("select-component");
    const placeholderElement = getByTestId("select-placeholder");

    expect(selectElement).toBeInTheDocument();
    expect(placeholderElement).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });

  it("render with option empty array", () => {
    const { getByTestId, asFragment, queryAllByRole } = render(
      <Select
        name="test-select"
        options={[]}
      />
    );
    const selectElement = getByTestId("select-component");
    expect(selectElement).toBeInTheDocument();
    const placeholderElement = getByTestId("select-placeholder");
    expect(placeholderElement).toBeInTheDocument();
    const options = queryAllByRole("option");
    expect(options).toHaveLength(0);
    expect(asFragment()).toMatchSnapshot();
  });

  it("multiple type", () => {
    const { getByTestId, asFragment } = render(
      <Select
        name="test-select-multiple"
        options={options}
        value={["value1"]}
        multiple
      />
    );

    const selectedComponent = getByTestId("select-component");
    expect(selectedComponent).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
});
