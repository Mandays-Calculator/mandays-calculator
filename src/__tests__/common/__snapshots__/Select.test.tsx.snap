// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Component multiple type 1`] = `
<DocumentFragment>
  <fieldset
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="select-component"
    >
      <div
        aria-controls=":r2:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="test-select-multiple test-select-multiple"
        class="MuiSelect-select MuiSelect-outlined MuiSelect-multiple MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="test-select-multiple"
        role="combobox"
        tabindex="0"
      >
        <div
          class="MuiBox-root css-yi3mkw"
        >
          <div
            class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root"
          >
            <span
              class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
            />
          </div>
        </div>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value="value1"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-ihdtdm"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </fieldset>
</DocumentFragment>
`;

exports[`Select Component render with option empty array 1`] = `
<DocumentFragment>
  <fieldset
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="select-component"
    >
      <div
        aria-controls=":r1:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="test-select test-select"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="test-select"
        role="combobox"
        tabindex="0"
      >
        <li
          aria-disabled="true"
          class="MuiButtonBase-root MuiMenuItem-root Mui-disabled MuiMenuItem-gutters Mui-disabled MuiMenuItem-root Mui-disabled MuiMenuItem-gutters css-15rbj0b-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="select-placeholder"
          role="menuitem"
          tabindex="-1"
        >
          <em>
            Please Select
          </em>
        </li>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-ihdtdm"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </fieldset>
</DocumentFragment>
`;

exports[`Select Component renders select without crashing 1`] = `
<DocumentFragment>
  <fieldset
    class="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      data-testid="select-component"
    >
      <div
        aria-controls=":r0:"
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="test-select test-select"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="test-select"
        role="combobox"
        tabindex="0"
      >
        <li
          aria-disabled="true"
          class="MuiButtonBase-root MuiMenuItem-root Mui-disabled MuiMenuItem-gutters Mui-disabled MuiMenuItem-root Mui-disabled MuiMenuItem-gutters css-15rbj0b-MuiButtonBase-root-MuiMenuItem-root"
          data-testid="select-placeholder"
          role="menuitem"
          tabindex="-1"
        >
          <em>
            Please Select
          </em>
        </li>
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
        tabindex="-1"
        value=""
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-ihdtdm"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </legend>
      </fieldset>
    </div>
  </fieldset>
</DocumentFragment>
`;
